### PR TITLE
add cli doc page

### PR DESCRIPTION
**Pull request by mooreds**

### CODE REVIEW

File: site/docs/v1/tech/admin-guide/cli.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+```
+npm i -g @fusionauth/cli
+```
+
+
+Then you can run commands.
+
+```
+fusionauth -k <APIKEY> theme:download <themeid>
```
Comment (mark-robustelli) [inline site/docs/v1/tech/admin-guide/cli.md:25]: @mooreds Trying to go through example.  I get a "unknown option -k" here. I tried the --help but only see -V and -h switches.  Am I missing something?

**Review by mark-robustelli — COMMENTED** [review]
Can't go much further till I get past the unknown option -k issue.

File: site/docs/v1/tech/admin-guide/cli.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+To learn more about the commands, use the `--help` switch.
+
+```
+fusionauth --help
+```
+
+## Functionality
+
+This tool allows you to easily retrieve and publish FusionAuth configuration from the command line.
```
Comment (mark-robustelli) [inline site/docs/v1/tech/admin-guide/cli.md:38]:
```suggestion
This tool allows you to easily retrieve and publish FusionAuth configurations from the command line.
```

File: site/docs/v1/tech/admin-guide/cli.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+* emails (coming soon)
+* lambdas (coming soon)
+
+The CLI is designed to work with complex version controlled configuration and includes support for localized content.
+
+## Source Code
+
+The FusionAuth CLI is open source.
+
+You can view the source code here: https://github.com/FusionAuth/fusionauth-node-cli
```
Comment (mark-robustelli) [inline site/docs/v1/tech/admin-guide/cli.md:52]:
```suggestion
You can view the source code here: [https://github.com/FusionAuth/fusionauth-node-cli](https://github.com/FusionAuth/fusionauth-node-cli)
```

File: site/docs/v1/tech/admin-guide/cli.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+* themes
+* emails (coming soon)
+* lambdas (coming soon)
+
+The CLI is designed to work with complex version controlled configuration and includes support for localized content.
+
+## Source Code
+
+The FusionAuth CLI is open source.
+
```
Comment (mark-robustelli) [inline site/docs/v1/tech/admin-guide/cli.md:51]: These do not show up as links on my page.  I am not sure if you want to show the link itself or do something like 

You can view the source code [here](https://github.com/FusionAuth/fusionauth-node-cli).

File: site/docs/v1/tech/admin-guide/cli.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+You can view the source code here: https://github.com/FusionAuth/fusionauth-node-cli
+
+You can view the npm package here: https://www.npmjs.com/package/@fusionauth/cli
```
Comment (mark-robustelli) [inline site/docs/v1/tech/admin-guide/cli.md:54]: Link not active as above

File: site/docs/v1/tech/admin-guide/cli.md (hunk 1 above)
Comment (mooreds) [reply site/docs/v1/tech/admin-guide/cli.md]: Yes, I think I put the args in the wrong order. Just pushed updated doco that should work.

File: site/docs/v1/tech/admin-guide/cli.md (hunk 5 above)
Comment (mooreds) [reply site/docs/v1/tech/admin-guide/cli.md:56]: I got spoiled by asciidoc, which does that automatically. Will make them links.

File: site/docs/v1/tech/admin-guide/cli.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+
+Then you can run commands.
+
+```
+fusionauth theme:download -k <APIKEY> <themeid>
+
+# modify your theme
+
+fusionauth theme:upload -k <APIKEY> <themeid>
```
Comment (mark-robustelli) [inline site/docs/v1/tech/admin-guide/cli.md:29]: I think there needs to be a little more description here of where the files need to be in order to upload.  I would assume that if I downloaded with the theme_id, they would be in a state to upload.  However the upload does not work for the theme id.  

Uploading theme acd1e588-236a-400c-bd35-16a38bbcaf46 from ./tpl/
Error uploading theme acd1e588-236a-400c-bd35-16a38bbcaf46:
ENOENT: no such file or directory, scandir './tpl/'

**Review by mark-robustelli — COMMENTED** [review]
Other changes look good, I can download the theme now but could use a little more description on how the upload works.

File: site/docs/v1/tech/admin-guide/cli.md (hunk 6 above)
Comment (mooreds) [reply site/docs/v1/tech/admin-guide/cli.md:29]: Thanks @mark-robustelli , that's very helpful. Will see what we need to do to have it work transparently.

Comment (mooreds) [reply site/docs/v1/tech/admin-guide/cli.md:29]: Hmm. this works fine for me on the sandbox:

```
dan@mooreds cli % fusionauth theme:download -k 90d8fb62-6f13-47d4-8ef6-1c3e687883c6 -h https://sandbox.fusionauth.io cccc4d5d-4317-4c71-9e3c-2302c414380b
  _____          _                  _         _   _     
 |  ___|   _ ___(_) ___  _ __      / \  _   _| |_| |__  
 | |_ | | | / __| |/ _ \| '_ \    / _ \| | | | __| '_ \ 
 |  _|| |_| \__ \ | (_) | | | |  / ___ \ |_| | |_| | | |
 |_|   \__,_|___/_|\___/|_| |_| /_/   \_\__,_|\__|_| |_|
                                                        
Downloading theme cccc4d5d-4317-4c71-9e3c-2302c414380b to ./tpl/
Theme cccc4d5d-4317-4c71-9e3c-2302c414380b downloaded to ./tpl/
dan@mooreds cli % ls
tpl
dan@mooreds cli % fusionauth theme:upload -k 90d8fb62-6f13-47d4-8ef6-1c3e687883c6 -h https://sandbox.fusionauth.io cccc4d5d-4317-4c71-9e3c-2302c414380b  
  _____          _                  _         _   _     
 |  ___|   _ ___(_) ___  _ __      / \  _   _| |_| |__  
 | |_ | | | / __| |/ _ \| '_ \    / _ \| | | | __| '_ \ 
 |  _|| |_| \__ \ | (_) | | | |  / ___ \ |_| | |_| | | |
 |_|   \__,_|___/_|\___/|_| |_| /_/   \_\__,_|\__|_| |_|
                                                        
Uploading theme cccc4d5d-4317-4c71-9e3c-2302c414380b from ./tpl/
Theme cccc4d5d-4317-4c71-9e3c-2302c414380b was uploaded successfully
```

Can you show me what issues you are seeing?

Comment (mark-robustelli) [reply site/docs/v1/tech/admin-guide/cli.md:29]: I'm not going to lie, just tried it again and it is working fine.  Very weird.  I spent a little time trying to re-create.  The only way I can is to delete the /tpl directory, so I must have done that without realizing.  Apologies.cl

Screenshot 2023-07-27 at 5 11 28 PM

Comment (mooreds) [reply site/docs/v1/tech/admin-guide/cli.md:29]: Ah, that makes sense.

**Comment by bradmccarty** [comment]
Good by me @mooreds

**Pull request merged**